### PR TITLE
GitHub Actions: Add support for --irc-channels

### LINKED ISSUE #423: GitHub Actions backend: support --irc-channels
There are a variety of existing GitHub Actions ([example](https://github.com/marketplace/actions/notify-irc)) that allow notifying an IRC channel based on various events. We should explore this and find a combination of events that gives us roughly equivalent functionality to what `--irc-channels` does for the Travis backend.

### PR DESCRIPTION
**Pull request by phadej**
This leverages https://github.com/marketplace/actions/irc-message-action
to send IRC notifications. Since you have to manually specify which events
trigger notifications, I'm restricting this to simple "The build passed"
and "The build failed" messages, but it would be possible to extend this to
other events later.

Fixes #423.

---

This is #432, rebased (conflict resolved) and squashed.

### CODE REVIEW

File: src/HaskellCI/GitHub.hs [hunk 1]
```diff
@@ -394,23 +394,31 @@ makeGitHub _argv config@Config {..} prj jobs@JobVersions {..} = do
...
-                | compiler@(GHC v) <- reverse $ toList versions
-                ]
-            }
+        , ghJobs = Map.fromList $ buildList $ do
+            item (mainJobName, GitHubJob
+                { ghjName            = "Haskell-CI Linux - GHC ${{ matrix.ghc }}"
+                , ghjRunsOn          = "ubuntu-18.04" -- TODO: use cfgUbuntu
+                , ghjNeeds           = []
+                , ghjSteps           = steps
+                , ghjIf              = Nothing -- TODO: Use cfgIrcIfInOriginRepo
```
Comment (RyanGlScott) [inline src/HaskellCI/GitHub.hs:403]: Should we file an issue about this remaining task?

**Review by RyanGlScott — COMMENTED** [review]
Thanks!

File: src/HaskellCI/GitHub.hs (hunk 1 above)
Comment (RyanGlScott) [reply src/HaskellCI/GitHub.hs:403]: Ah, I think this was done in #436.

Comment (phadej) [reply src/HaskellCI/GitHub.hs:403]: Yeah, writing a patch felt easier than opening an issue.

**Pull request merged**